src/cr/cube/cubepart.py

```diff
@@ -505,7 +511,43 @@ def row_percentages(self):
 
     @lazyproperty
     def row_proportions(self):
-        return np.array([row.proportions for row in self._matrix.rows])
+        """2D np.float64 ndarray of row-proportion for each matrix cell.
+
+        Each row-proportion value is that cell's fraction of the weighted-N for the row
+        (aka. rows_margin), a number between 0.0 and 1.0.
+        """
+        rows_margin = self.rows_margin
+        with np.errstate(divide="ignore", invalid="ignore"):
+            # --- numpy broadcasting only works "vertically", so in the 1D margin case,
+            # --- we need to "rotate" (transpose) the counts such that rows are running
+            # --- "vertically", and then re-transpose the result to get back
+            # --- (rows, cols) orientation.
+            if rows_margin.ndim == 1:
+                return (self.counts.T / self.rows_margin).T
```

Comment:
use just `rows_margin`, not the `self.rows_margin`, if we instantiated it already.

Reply:
yep, good catch